router/middleware/settings.go

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package middleware
+
+import (
+	"context"
+
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+
+	"github.com/go-vela/server/database"
+	"github.com/go-vela/server/router/middleware/settings"
+)
+
+// Settings is a middleware function that fetches the latest settings and
+// attaches to the context of every http.Request.
+func Settings(d database.Interface) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		s, err := d.GetSettings(context.Background())
```

Comment:
can we optimize this? like, what if the API call doesnt NEED settings *or any of the other resources?* 


we are definitely over-querying the database